crates/orderbook/src/database/orders.rs

```diff
@@ -220,56 +220,95 @@ const ORDERS_FROM: &str = "\
-            .execute(&self.pool)
-            .await
-            .map(|_| ())
-            .map_err(|err| {
-                if let sqlx::Error::Database(db_err) = &err {
-                    if let Some(Cow::Borrowed("23505")) = db_err.code() {
-                        return InsertionError::DuplicatedRecord;
-                    }
+    let receiver = order
+        .order_creation
+        .receiver
+        .map(|address| address.as_bytes().to_vec());
+    sqlx::query(QUERY)
+        .bind(order.order_meta_data.uid.0.as_ref())
+        .bind(order.order_meta_data.owner.as_bytes())
+        .bind(order.order_meta_data.creation_date)
+        .bind(order.order_creation.sell_token.as_bytes())
+        .bind(order.order_creation.buy_token.as_bytes())
+        .bind(receiver)
+        .bind(u256_to_big_decimal(&order.order_creation.sell_amount))
+        .bind(u256_to_big_decimal(&order.order_creation.buy_amount))
+        .bind(order.order_creation.valid_to)
+        .bind(&order.order_creation.app_data.0[..])
+        .bind(u256_to_big_decimal(&order.order_creation.fee_amount))
+        .bind(DbOrderKind::from(order.order_creation.kind))
+        .bind(order.order_creation.partially_fillable)
+        .bind(&*order.order_creation.signature.to_bytes())
+        .bind(DbSigningScheme::from(
+            order.order_creation.signature.scheme(),
+        ))
+        .bind(order.order_meta_data.settlement_contract.as_bytes())
+        .bind(DbSellTokenSource::from(
+            order.order_creation.sell_token_balance,
+        ))
+        .bind(DbBuyTokenDestination::from(
+            order.order_creation.buy_token_balance,
+        ))
+        .bind(u256_to_big_decimal(&order.order_meta_data.full_fee_amount))
+        .execute(transaction)
+        .await
+        .map(|_| ())
+        .map_err(|err| {
+            if let sqlx::Error::Database(db_err) = &err {
+                if let Some(Cow::Borrowed("23505")) = db_err.code() {
+                    return InsertionError::DuplicatedRecord;
                 }
-                InsertionError::DbError(err)
+            }
+            InsertionError::DbError(err)
+        })
+}
+
+async fn insert_fee(
+    uid: &OrderUid,
+    fee: &FeeParameters,
+    transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+) -> Result<(), InsertionError> {
+    const QUERY: &str = "\
+                INSERT INTO order_fee_parameters (\
+                    order_uid, gas_amount, gas_price, sell_token_price) \
+                VALUES($1, $2, $3, $4)\
+            ;";
+    sqlx::query(QUERY)
+        .bind(uid.0.as_ref())
+        .bind(fee.gas_amount)
+        .bind(fee.gas_price)
+        .bind(fee.sell_token_price)
+        .execute(transaction)
+        .await
+        .map(|_| ())
+        .map_err(InsertionError::DbError)
+}
+
+#[async_trait::async_trait]
+impl OrderStoring for Postgres {
+    async fn insert_order(&self, order: &Order, fee: FeeParameters) -> Result<(), InsertionError> {
```

Comment:
Any reason why we don't have `FeeParameters` added as a field to `Order` struct? Not objecting to it, just wondering if you considered it if you did why you chose against it?

Reply:
Nevermind. This would only make sense if we someday intended on returning this as part of an order, which AFAIK is not the plan.